views/js/qtiCreator/widgets/interactions/textEntryInteraction/states/Map.js

```diff
@@ -56,7 +56,7 @@ define([
                     score = parseInt($tr.find('.score').val());
 
                 //@todo trigger validate score to check if is a valid score
-                if(!isNaN(score)){
+                if(!isNaN(score) && !(value in response.mapEntries)){
```

Comment:
The `in` operator is going to check property in the prototype chain as well 
```suggestion
                if(!isNaN(score) && !(response.mapEntries.hasOwnProperty(value)){
```